python/ml4ir/model/ranking_model.py

```diff
@@ -73,7 +73,7 @@ def __init__(
             )
 
             # Define metrics
-            metrics: List[str] = [
+            metrics: List[Type[kmetrics.Metric]] = [
```

Comment:
why does this need too be `List[Type[kmetrics.Metric]]` and not just `List[kmetrics.Metric]`, do you know?

Reply:
It is `Type[Metric]` because it is a class and not instantiated yet.

Reply:
it's a class and not instantiated yet?  I... don't understand that.  It's returning a function which returns a Metric?  Is that what you mean?